front/src/common/api/osrdEditoastApi.ts

```diff
@@ -2382,7 +2340,142 @@ export type Operation =
     } & {
       operation_type: 'DELETE';
     });
+export type ObjectRef = {
+  obj_id: string;
+  type: ObjectType;
+};
+export type InfraErrorType =
+  | {
+      error_type: 'duplicated_group';
+      original_group_path: string;
+    }
+  | {
+      error_type: 'empty_object';
+    }
+  | {
+      error_type: 'invalid_group';
+      group: string;
+      switch_type: string;
+    }
+  | {
+      error_type: 'invalid_reference';
+      reference: ObjectRef;
+    }
+  | {
+      error_type: 'invalid_route';
+    }
+  | {
+      error_type: 'invalid_switch_ports';
+    }
+  | {
+      error_type: 'missing_route';
+    }
+  | {
+      endpoint: Endpoint;
+      error_type: 'missing_buffer_stop';
+    }
+  | {
+      error_type: 'node_endpoints_not_unique';
+    }
+  | {
+      error_type: 'object_out_of_path';
+      reference: ObjectRef;
+    }
+  | {
+      error_type: 'odd_buffer_stop_location';
+    }
+  | {
+      error_type: 'out_of_range';
+      expected_range: number[];
+      position: number;
+    }
+  | {
+      error_type: 'overlapping_electrifications';
+      reference: ObjectRef;
```

Comment:
In practice, it will always be an `Electrification`. Returning the object type makes the response more explicit and easier for the front-end to use. The front-end doesn't need to check the error type to create the link to the object.